### ISSUE TITLE
Building an apk from a cordova project or building a straight release apk will always be corrupt r50, r51

### ISSUE THREAD

**Issue by drexegar** [Scirra/Construct-3-bugs#746]
## Problem description

Building an app on android gives me a corrupted APK file

## Attach a .c3p

https://drive.google.com/file/d/0ByybatBADJBNQVZobHVHTHA2TlU/view?usp=sharing

## Steps to reproduce

1. Export with Cordova
2. Build Cordova Project
2. Go to export manager and Build APK from Cordova Project
3. Test on Android
4. Android gives a message "The Package appears to be corrupt".

5. When Building a straight release APK, it will also be corrupt

## Observed result

I ended up with a corrupted apk every time from a cordova project in the project manager, and when I export a release apk, that will be corrupted too.

## Expected result

Have a working APK

## Effect Browsers

* Chrome: yes

## System details

*If you see a crash report dialog, please copy and paste it to where it says "PASTE HERE" below. Otherwise please go to **Menu** -> **About** -> **Platform information** and paste that information there instead.*

<details><summary>View details</summary>

PASTE HERE

</details>

**Comment by shortercode** [comment]
Are you seeing this issue only on r50 or does it happen on r49 as well?

**Comment by drexegar** [comment]
I have no problem building apps in r49 it started with r50.

**Comment by shortercode** [comment]
Odd, I don't seem to have any issue building a working APK with your project. Can you detail the exact menus and settings you go through when you build.

**Comment by drexegar** [comment]
ok let me try again after i fix the save problem will report in a bit.

**Comment by drexegar** [comment]
SORRY I UPLOADED WRONG FILE HERE THE LATEST FILE SAVED IN r50
https://drive.google.com/file/d/0ByybatBADJBNQVZobHVHTHA2TlU/view?usp=sharing

I still get a corrupted APK with this one, I will also update the post

By the way this file was fixed with the "work around" for the saving problem

**Comment by shortercode** [comment]
I'm sorry that save is still building and installing correctly for me, there hasn't been any changes to the exporters for several releases so I'm surprised that r50 is specifically causing you problems. r51 has gone out now, so I'm curious if it still breaks for you on r51?

**Comment by drexegar** [comment]
wow. Thats very weird! I just build it right when I posted. ok then ill test r51!

**Comment by drexegar** [comment]
Ok I made a huge mistake I  messed up the step, I can build and apk, but if I build a cordova project and then build that into an apk, that apk will get corrupted. Will update the steps.

**Comment by drexegar** [comment]
OK after doing more testing, 

If I export a release app, it will be corrupt
If I bulid a app from a cordova project saved on the project mananger, it will be corrupt

If I export a debug app, it will work. r51.

**Comment by shortercode** [comment]
Okay I haven't been testing release builds, only debug ones. I presume your performing apk signing and alignment before installing?

**Comment by drexegar** [comment]
Actually no, I have no clue yet what to do when performing an apk signing and alignment. I'll look that up.

**Comment by shortercode** [comment]
Ah okay so basically when you build a debug build the APK it's labelled as a debug build, and a device which has developer mode enabled will happily install it as a debug build for testing.

However, with a release build before a device will allow it to be installed the APK must be signed with a developer certificate. Aligning the file is typically done at this stage as well, it involves moving the files around in the APK to speed up file reads when the app is running

**Comment by shortercode** [comment]
Unless your actually publishing your game debug builds are the way to go :)

**Comment by drexegar** [comment]
yes I want to publish my game, I guess i will have to look it up!

**Comment by drexegar** [comment]
Im pretty sure there are no problems, but here is a minimal c3p file:
https://drive.google.com/file/d/0ByybatBADJBNOUppb180em9RRDg/view?usp=sharing

Its works on browser, but will not work as an apk, touch or start on layout is ignored, as well as the autoplay.

Will update the post too.

**Comment by shortercode** [comment]
Seems to me touch is working correctly in the APK, 2 issues I can see:

1. There's no user gesture to start the video playing, so it stays in the paused state forever
2. When built as an APK it cannot load the video, I think this is because your using an upper case `I` to reference the file in the video object, but the filename is lower ( some systems are case sensitive )

There's some notes on issues platform compatibility in the [video documentation](https://www.construct.net/gb/make-games/manuals/construct-3/plugin-reference/video)

**Comment by shortercode** [comment]
I'm going to close this bug, from our conversation I think it's just because you were trying to install unsigned release APKs. If there is an issue then let me know and I'll reopen it
